File: app/src/main/java/com/temperoni/recipes/ui/models/InstructionItemViewModel.java
package com.temperoni.recipes.ui.models;

import java.util.List;

/**
 * @author Leandro Temperoni
 */
public class InstructionItemViewModel {

    private String description;
    private boolean descriptionAvailable;
    private List<String> steps;

    public String getDescription() {
        return description;
    }

    public boolean hasDescriptionAvailable() {
        return descriptionAvailable;
    }

    public List<String> getSteps() {
        return steps;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setDescriptionAvailable(boolean descriptionAvailable) {
        this.descriptionAvailable = descriptionAvailable;
    }

    public void setSteps(List<String> steps) {
        this.steps = steps;
    }
}

File: app/src/main/java/com/temperoni/recipes/ui/models/InstructionItemViewModel.kt
package com.temperoni.recipes.ui.models

/**
 * @author Leandro Temperoni
 */
class InstructionItemViewModel {

    var description: String? = null
    private var descriptionAvailable: Boolean = false
    var steps: List<String>? = null

    fun hasDescriptionAvailable(): Boolean {
        return descriptionAvailable
    }

    fun setDescriptionAvailable(descriptionAvailable: Boolean) {
        this.descriptionAvailable = descriptionAvailable
    }
}
